### ISSUE TITLE
Inference output determined by batch size

### ISSUE THREAD

**Issue by levelvc** [vonclites/squeezenet#1]
Hi Vonclites, thanks for sharing the squeezenet implementation. I was playing around with it and I noticed that the output to the inference on eval is dependent on the batch size of the input. For instance, with a batch size of 1 the output it different than batch size of 5. Looking into it, but perhaps you may have seen this.

**Comment by vonclites** [comment]
Ahhh, my first guess is this is due to how epochs are calculated combined with how the looping over the data is done.  When you use a batch size of 1, the # of epochs are perfectly calculated to run over exactly all your data once.  However with a batch size > 1, then the number of epochs is the ceiling of dividing the size of your training set by the batch size.  For example, with 42 test samples and a batch size of 5, the number of epochs would be ceil(42 / 5) = 9.  But 9 * 5 is 45 samples, so you'd be evaluating 3 test samples twice....

It's a lazy way to calculate epochs and do the looping...
Really, the evaluation_loop should make better use of the [batch queue](https://www.tensorflow.org/api_docs/python/io_ops/input_pipeline#batch).

Glad you are making use of the model though!  Just beware... this was quickly put together for a class homework and definitely is not polished.

**Comment by levelvc** [comment]
Thanks! Evaluation loop aside, if I load the checkpoint train model directly and run inference directly, the logit outputs are wildly different for various batch sizes with same input. Just wondering why that would be? 

Yes thanks was just playing around with squeezenet.

**Comment by levelvc** [comment]
Hey, so it seems the issue is that `batch_norm` needs to be set to `is_training=False` during inference. You can do this by adjusting your squeezenet model as such:

`def squeezenet_arg_scope(is_training=True):
    batch_norm_params = {'is_training': is_training, 'decay': 0.9, 'updates_collections': None}
    with slim.arg_scope([slim.conv2d], normalizer_fn=slim.batch_norm, normalizer_params=batch_norm_params) as sc:
        return sc`

**Comment by vonclites** [comment]
Ohhh... woops! Good call, thank you!

**Comment by vonclites** [comment]
Finally had some time to include your fix. Thanks again.

**Comment by Bhagyoday-Patil** [comment]
Hi @levelvc @vonclites 
I have followed similar steps
Converting my images to TFrecords and then trying to use the train_squeezenet script for the same
When i do the same I am getting the below log.txt error

[log.txt](https://github.com/vonclites/squeezenet/files/2529546/log.txt)

Please let me know what exactly is wrong and the workaround for it

Thanks

**Comment by aazout** [comment]
Looks like your batch size is zero.

**Comment by Bhagyoday-Patil** [comment]
> Looks like your batch size is zero.
> […](#)
> On Oct 30, 2018, at 5:16 AM, Bhagyoday-Patil ***@***.***> wrote: Hi @levelvc @vonclites I have followed similar steps Converting my images to TFrecords and then trying to use the train_squeezenet script for the same When i do the same I am getting the below log.txt error log.txt Please let me know what exactly is wrong and the workaround for it Thanks — You are receiving this because you were mentioned. Reply to this email directly, view it on GitHub, or mute the thread.

Sorry @levelvc slight mistake
Please find attachment logs.
[logs.txt](https://github.com/vonclites/squeezenet/files/2532737/logs.txt)

I have used [https://github.com/kwotsin/create_tfrecords](url)  to convert my images to tfrecords. 
The same tfrecords when given as input to the train_squeezenet script is throwing that error. I have only 2 classes. I am not understanding what exactly is wrong. I have not made any changes in any of the scripts.
tensorflow version = 1.11.0
ubuntu version = 16.04.3
no GPU/CUDA

**Comment by Bhagyoday-Patil** [comment]
@vonclites @muthiyanbhushan Can you please help me with this error.
As descried above I am getting this error.
PFA [log_1.7.1.txt](https://github.com/vonclites/squeezenet/files/2541354/log_1.7.1.txt)
